app/Http/Controllers/PostController.php

```diff
@@ -81,30 +79,24 @@ public function getUserPost()
 	 */
 	public function getAllPost(Request $request)
 	{
-		if(!$request->offer_or_ask){
-			$posts = $this->post->all();
-			foreach ($posts as $post) {
-				$post->user;
-			}
-
-			return $posts;
-		}
+		$offset 		= ($request->offset) ?  $request->offset : 0;
+		$posts  		= (!$request->offer_or_ask) ? 
+				  		  $this->post :
+				          $this->post->where('offer_or_ask',$request->offer_or_ask);
+		$postAfterSkip  = $posts->skip($offset)
+								->take(config('constants.POST_SIZE'))->get();
 
-		$choice = ($request->offer_or_ask == self::OFFER) ? 
-				  self::OFFER : self::ASK;
-		$posts = $this->post->where('offer_or_ask',$choice)->get();
+		ImageHelper::includeImageUserInPost($postAfterSkip);
 		
-		foreach ($posts as $post) {
-			$post->user;
-			if($choice == self::OFFER){
-				$images = $this->image->wherePostId($post->id)->get();
-				$imageCollection = Helper::pushArray($images, 'image_location');
-				$post['images'] = $imageCollection;
-			}
+		$response 		= Helper::postResponse('0072', $postAfterSkip);
+		$response 		= array_merge($response,[
+						 'message' => sprintf($response['message'],$posts->count()),
```

Comment:
don't give priority to array_merge()
what's the point in merging array with itself?
$response['message] = ...